feat: Add github_organization_network_configuration resource

austenstone opened the pull request:
## Overview

This PR adds a new `github_organization_network_configuration` resource to manage GitHub organization network configurations for hosted compute services.

It also adds the missing linkage for GitHub-hosted private networking on `github_actions_runner_group` via `network_configuration_id`, so hosted runners can consume private networking through their assigned runner group.

## What this PR adds

- New resource: `github_organization_network_configuration`
- Full CRUD support for organization network configurations
- Import support for organization network configurations
- Support for `github_actions_runner_group.network_configuration_id`
- Acceptance coverage for:
  - create, update, and import flows for `github_organization_network_configuration`
  - create, update, import, and clear flows for runner group network configuration association
- Documentation updates for:
  - `github_organization_network_configuration`
  - `github_actions_runner_group`
  - `github_actions_hosted_runner`
- Example updates for hosted runners and public IP configuration

## Schema

### `github_organization_network_configuration`

- `name` (Required) - Network configuration name (1-100 chars, alphanumeric plus `._-`)
- `compute_service` (Optional) - Hosted compute service: `none` or `actions` (default: `none`)
- `network_settings_ids` (Required) - Exactly one network settings ID
- `created_on` (Computed) - Creation timestamp
- `id` (Computed) - Resource ID

### `github_actions_runner_group`

- `network_configuration_id` (Optional) - The ID of the hosted compute network configuration associated with the runner group for GitHub-hosted private networking

## Notes

- This resource is organization-scoped and validates organization mode via `checkOrganization()`.
- GitHub private networking for GitHub-hosted runners is configured on the runner group, not directly on `github_actions_hosted_runner`.
- For Azure private networking, `network_settings_ids` must use the GitHub-facing network settings identifier (`GitHubId`), not the Azure ARM resource ID.
- Network configurations are available for GitHub Enterprise Cloud organizations.

## Validation

- `go test ./github -run TestAccGithubOrganizationNetworkConfiguration -count=1`
- `go test ./github -run TestAccGithubActionsRunnerGroup -count=1`
- `go test ./github -run '^$' -count=1`

deiga commented on github/resource_github_organization_network_configuration.go:
```diff
@@ -0,0 +1,193 @@
+
+	d.SetId(configuration.GetID())
+
+	return resourceGithubOrganizationNetworkConfigurationRead(d, meta)
```
Please don't call any CRUD function directly. All fields need to be set directly in Create

deiga commented on github/resource_github_organization_network_configuration.go:
```diff
@@ -0,0 +1,193 @@
+	configuration, resp, err := client.Organizations.GetNetworkConfiguration(ctx, orgName, networkConfigurationID)
+	if err != nil {
+		if ghErr, ok := err.(*github.ErrorResponse); ok && ghErr.Response.StatusCode == http.StatusNotFound {
+			log.Printf("[WARN] Removing organization network configuration %s from state because it no longer exists in GitHub", networkConfigurationID)
```
Please use tflog

austenstone commented:
Thanks — I addressed the review feedback and pushed follow-up changes.

Summary of updates:

- switched the resource to context-aware CRUD functions
- added the top-level resource description
- updated `Create` to set state directly instead of calling another CRUD function
- replaced logging with `tflog`
- aligned acceptance tests to use `ConfigStateChecks`
- added docs/error-handling guidance for the Azure network settings scope gotcha that came up in live validation

I also re-tested the resource end-to-end against a real org/Azure-backed setup after the refactor.

Would appreciate another pass when you have a chance.

austenstone commented:
Healthy from my side ✅

I re-tested the full flow and everything is behaving as expected: create/update/destroy all work cleanly, CI is green, and the acceptance coverage passes. I also validated the real-world integration path with org network configurations, runner group association, and hosted runner private networking.

deiga commented on github/resource_github_actions_runner_group.go:
```diff
@@ -186,19 +254,14 @@ func resourceGithubActionsRunnerGroupCreate(d *schema.ResourceData, meta any) er
-			return runnerGroup, resp, nil
+	if networkConfigurationID, ok := d.GetOk("network_configuration_id"); ok {
+		networkConfigurationIDValue := networkConfigurationID.(string)
+		if _, err = updateOrganizationRunnerGroupNetworking(client, ctx, orgName, runnerGroup.GetID(), &networkConfigurationIDValue); err != nil {
```
This is the Create function, if it needs to call multiple API endpoints please add comments explaining the logic

austenstone commented:
closing in favor of #3274
